src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,111 @@
+  name: { column: 'Name', sortable: true },
+  sex: { column: 'Sex', sortable: true },
+  born: { column: 'Born', sortable: true },
+  died: { column: 'Died', sortable: true },
+  motherName: { column: 'Mother', sortable: false },
+  fatherName: { column: 'Father', sortable: false },
+};
+
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+export const PeopleTable = ({
+  people,
+  selectedPerson,
+  peopleMap,
+  onArrowClick,
+  sortColumn,
+  sortDirection,
+}: {
+  people: Person[];
+  selectedPerson?: string | null;
+  peopleMap: PeopleMap;
+  onArrowClick: (column: string) => SearchParams;
+  sortColumn: keyof Person | null;
+  sortDirection: boolean;
+}) => {
+  const shouldHighlightUser = (slug: string) =>
+    selectedPerson === slug ? 'has-background-warning' : '';
+  // TODO class needs to receive the sort
+
   return (
     <table
       data-cy="peopleTable"
       className="table is-striped is-hoverable is-narrow is-fullwidth"
     >
       <thead>
         <tr>
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Name
-              <a href="#/people?sort=name">
-                <span className="icon">
-                  <i className="fas fa-sort" />
-                </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Sex
-              <a href="#/people?sort=sex">
-                <span className="icon">
-                  <i className="fas fa-sort" />
-                </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Born
-              <a href="#/people?sort=born&amp;order=desc">
-                <span className="icon">
-                  <i className="fas fa-sort-up" />
-                </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Died
-              <a href="#/people?sort=died">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+          {Object.entries(TABLE_COLUMNS).map(([key, { column, sortable }]) => {
+            const isSortedColumn = sortColumn === key;
+
+            const sortIcon = isSortedColumn
+              ? sortDirection
+                ? 'fa-sort-down'
+                : 'fa-sort-up'
+              : 'fa-sort';
+
+            return (
+              <th key={column}>
+                <span
+                  className="is-flex is-flex-wrap-nowrap"
+                  onClick={sortable ? () => onArrowClick(key) : undefined}
+                  style={{ cursor: sortable ? 'pointer' : 'default' }}
+                >
+                  {column}
+                  {sortable && (
+                    <SearchLink params={onArrowClick(column.toLowerCase())}>
```

Comment:
The `onArrowClick` function is called twice: once directly in the `onClick` handler and once within the `SearchLink` component. This might lead to unexpected behavior. Consider removing the direct call in the `onClick` handler if the `SearchLink` is handling the sorting logic.